### ISSUE TITLE
Allow Field Types/Format List to Be Supplied

### ISSUE THREAD

**Issue by spencerfinnell** [berlindb/core#137]
I'm not exactly sure what this would look like, but I thought I would log it in case someone else runs in to something similar.

Working with Stripe objects, I have a table with a column `object_id`. It turns out WordPress [sets a default field type for `object_id`, and other column names](https://github.com/WordPress/WordPress/blob/9fb5112732c62a33d3e56d15fd402f88a97943fa/wp-includes/load.php#L591). This list of field types are used to prepare the SQL statement and format values accordingly when a field list and corresponding format list are not supplied during `$wpdb::insert` or `$wpdb::update` operations.

BerlinDB [does not pass a format list](https://github.com/berlindb/core/blob/da041d646d5349a0e263487bd01e8dc99477a185/src/Database/Query.php#L1834) when calling `$wpdb::insert( string $table, array $data, array|string $format = null )` causing the WordPress core list to be used. This was causing Stripe Object IDs, which are strings, to be converted to an integer.

My workaround was to name the column `_object_id` and shim the `insert` and `update` operations to add the `_` prefix. Other workarounds might be more involved and therefore maybe unnecessary, but again, just logging this for reference. Perhaps a format list could be auto generated based on [columns supplied to `Schema`](https://github.com/berlindb/core/blob/master/src/Database/Schema.php#L33)?

Semi-related: #124

**Comment by JJJ** [comment]
> My workaround was to name the column `_object_id` and shim the `insert` and `update` operations to add the `_` prefix.

Ouch! That's really annoying to have to do.

> Perhaps a format list could be auto generated based on [columns supplied to `Schema`](https://github.com/berlindb/core/blob/master/src/Database/Schema.php#L33)?

This is the way, IMO.

We should prioritize fixing this ASAP.

**Comment by JJJ** [comment]
The `Query` class needs a prettier (and more useful) version of:

```php
$patterns = array_intersect_key(
	wp_list_pluck( $this->get_columns(), 'pattern', 'name' ),
	$save,
);

```

**Comment by JJJ** [comment]
Realizing now that `pattern` probably should have been `format`.

Maybe we can alias this at a later date. Noting here for the future.

**Comment by JJJ** [comment]
Note: fixing #124 with them same PR imminently.

**Comment by JJJ** [comment]
Fixed for 2.1.0.
